pkg/build/buildkit/runner.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buildkit
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/moby/buildkit/client"
+	"github.com/okteto/okteto/pkg/analytics"
+	"github.com/okteto/okteto/pkg/env"
+	"github.com/okteto/okteto/pkg/log/io"
+)
+
+const (
+	// defaultMaxAttempts is the default number of attempts to try to do a build
+	defaultMaxAttempts = 3
+
+	// MaxRetriesForBuildkitTransientErrorsEnvVar is the environment variable to set the number of retries to wait for buildkit to be available
+	MaxRetriesForBuildkitTransientErrorsEnvVar = "OKTETO_MAX_RETRIES_FOR_BUILDKIT_TRANSIENT_ERRORS"
+)
+
+type registryImageChecker interface {
+	GetImageTagWithDigest(tag string) (string, error)
+}
+
+type buildkitWaiterInterface interface {
+	WaitUntilIsUp(ctx context.Context) error
+}
+
+type buildkitClientFactory interface {
+	GetBuildkitClient(ctx context.Context) (*client.Client, error)
+}
+
+type runnerMetadata struct {
+	attempts int
+	// TODO: use this to track the time it takes to solve a build instead of the full build time
```

Comment:
There is no ticket but checking the actual (code)[https://github.com/okteto/okteto/blob/master/cmd/build/v2/build.go#L293-L301] we are also adding the time waiting for the actual build and retries, instead of just the build proccess. With this change and if the retries augments this problem will be more common